datastore/importers/nmdb-import-systeminfo/Parser.test.cpp

```diff
@@ -0,0 +1,253 @@
+      DHCP Enabled:    Yes
+      IP address(es)
+      [01]: 192.168.0.2
+      [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+      )STR", //Mising first string
+      R"STR([01]: Realtek PCIe GbE Family Controller
+      Connection Name:
+      DHCP Enabled:    Yes
+      IP address(es)
+      [01]: 192.168.0.2
+      [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+      )STR", //Missing connection name
+      R"STR([01]: Realtek PCIe GbE Family Controller
+      Connection Name: Ethernet
+      DHCP Enabled:
+      IP address(es)
+      [01]: 192.168.0.2
+      [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+      )STR", //Missing dhcp enabled data
+      R"STR([01]: Realtek PCIe GbE Family Controller
+      Connection Name: Ethernet
+      DHCP Enabled:    Yes
+      [01]: 192.168.0.2
+      [02]: 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+      )STR", //Missing ipaddresses label
+      R"STR([01]: Realtek PCIe GbE Family Controller
+      Connection Name: Ethernet
+      DHCP Enabled:    Yes
+      IP address(es)
+      )STR" //Missing Ipaddress data
+    };
+    for (const auto& test : testsOk) {
+      BOOST_TEST(nmdp::test(test.c_str(), parserRule, blank),
+                "Parse rule 'network_card': " << test);
+    }
+    for (const auto& test: testsFail) {
+      BOOST_TEST(!nmdp::test(test.c_str(), parserRule, blank),
+                "Parse rule 'network_card fails': " << test);
+    }
+  }
+  {
+    const auto& parserRule {tp.ipAddressLine};
+    std::vector<std::string> testsOk {
+      R"STR(192.168.1.1)STR"
+    };
+
+    std::vector<std::string> testsFail {
+      R"STR(.168.1.1)STR", //Missing first octet
+      R"STR(192..1.1)STR",  //Missing second octet
+      R"STR(192.168..1)STR", //Missing vlan octet
+      R"STR(192.168.1.)STR", //Missing device octet
+      R"STR(192168.11)STR", //Only one period
+      R"STR(19216811)STR" //No periods
+    };
+    for (const auto& test : testsOk) {
+      BOOST_TEST(nmdp::test(test.c_str(), parserRule, blank),
+                "Parse rule 'ipAddressLine': " << test);
+    }
+    for (const auto& test: testsFail) {
+      BOOST_TEST(!nmdp::test(test.c_str(), parserRule, blank),
+                "Parse rule 'ipAddressLine fails': " << test);
+    }
+  }
+}
+
+BOOST_AUTO_TEST_CASE(testhotfixes)
+{
+  TestParser tp;
+  {
+    const auto& parserRule {tp.hotfixes};
+    std::vector<std::string> testsOk {
+      R"STR(Hotfix(s):           10 Hotfix(s) Installed.
+      [01]: KBXXXXXX
+      [02]: KBXXXXXX
+      [03]: KBXXXXXX
+      [04]: KBXXXXXX
+      [05]: KBXXXXXX
+      [06]: KBXXXXXX
+      [07]: KBXXXXXX
+      [08]: KBXXXXXX
+      [09]: KBXXXXXX
+      [10]: KBXXXXXX
+      )STR"
+    };
+    std::vector<std::string> testsFail {
+      R"STR(KBXXXXXX)STR", //Missing index
+      R"STR([01]: )STR"    //Missing Hotfix
+    };
+    for (const auto& test : testsOk) {
+      BOOST_TEST(nmdp::test(test.c_str(), parserRule, blank),
+                "Parse rule 'hotfixes': " << test);
+    }
```

Comment:
While you should add a check since state was changed, you don't need to check that each hotfix was added since that logic is technically in another rule.  Just check that the hotfix count is not zero (you probably also need to clear the `hotfixes` vector between runs).

Reply:
Okay I added the clear hotfix